### PR TITLE
hw19 (spring data, email)

### PR DESCRIPTION
**Pull request by Sofasmile**

### CODE REVIEW

File: src/main/java/com/spring/springdata/config/ApplicationConfig.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    private String password;
+    @Value("${hibernate.dialect}")
+    private String dialect;
+    @Value("${ddl-auto}")
+    private String ddlAuto;
+    @Value("${hibernate.show_sql}")
+    private String showSQL;
+
+    @Bean
+    public DriverManagerDataSource DataSource() {
```
Comment (Taniann) [inline src/main/java/com/spring/springdata/config/ApplicationConfig.java]: Method names are written in lowerCamelCase.

File: src/main/java/com/spring/springdata/servise/StudentService.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import java.util.Optional;
+
+public interface StudentService {
+    void saveStudent(Student student);
+
+    void updateStudent(Student student);
+
+    void deleteStudent(Long id);
+
+    Optional<Student> getStudentById(Long id);
```
Comment (Taniann) [inline src/main/java/com/spring/springdata/servise/StudentService.java]: Better return just student not Optional in Service

File: src/main/java/com/spring/springdata/servise/StudentServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    @Override
+    public void deleteStudent(Long id) {
+        studentRepository.deleteById(id);
+    }
+
+    @Override
+    public Optional<Student> getStudentById(Long id) {
+
```
Comment (Taniann) [inline src/main/java/com/spring/springdata/servise/StudentServiceImpl.java]: Remove redundant blank line

**Review by Taniann — CHANGES_REQUESTED** [review]
Please make a little fixing

File: src/main/java/com/spring/springdata/entity/Student.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import javax.validation.constraints.Email;
+import java.time.LocalDate;
+
+@Getter
+@Setter
+@Entity
+@NoArgsConstructor
+@Accessors(chain = true)
+@Table(name = "students")
+@EqualsAndHashCode(exclude = {"id"})
```
Comment (asulyma) [inline src/main/java/com/spring/springdata/entity/Student.java]: What the main purpose of this? I don't see any usages it for HashMap or HashSet

File: src/test/java/com/spring/springdata/SpringDataApplicationTests.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.time.LocalDate;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = ApplicationConfig.class)
```
Comment (Taniann) [inline src/test/java/com/spring/springdata/SpringDataApplicationTests.java:23]: for future: dont use your prod database for testing
use tests db

Comment (Sofasmile) [reply src/test/java/com/spring/springdata/SpringDataApplicationTests.java:23]: Ok, thank you)

**Pull request merged**